### PR TITLE
CDAP-18249 Uploading a directive does not close an upload window

### PR DESCRIPTION
**Pull request by itsankit-google**
[CDAP-18249](https://cdap.atlassian.net/browse/CDAP-18249)

When the user tries to upload a UDD, if the json had the attribute property missing, then a null pointer exception is raised while updating the artifact properties in the transaction runner which aborts the transaction at that point but the UI freezes at that point because it didn't return a meaningful error message rather the request was terminated with the status code 500 and response undefined.

Tested using this [directive](https://drive.google.com/drive/folders/1X2Scm_RuQBP8CpFsxgLSWIhTWpiKMCyU?resourcekey=0-EZL374Czlfp94LDXzr59fA&usp=sharing)  provided by the reporter.

Previous [PR](https://github.com/cdapio/cdap/pull/13660) for context.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandler.java [hunk 1]
```diff
@@ -341,6 +341,11 @@ public void writeProperties(FullHttpRequest request, HttpResponder responder,
       throw new BadRequestException("Unable to read properties from the request.", e);
     }
 
+    if (properties == null) {
+      throw new BadRequestException("Properties attribute is missing in the json or set as null, " +
+                                      "add properties: {} in json");
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandler.java]: The error message is unclear. This API expects the request body is a properties map, not an attribute inside. Maybe reword it as: `"Missing properties from the request. Please check that the request body is a json map from string to string."`

**Review by chtyim — APPROVED** [review]
The fix itself LGTM. Please fix the wordings before merging.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandler.java (hunk 1 above)
Comment (itsankit-google) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandler.java]: done
